sign-compare error in posix.cc

The following error occurs due to `-Werror` on gcc 9.1

```
tiledb/sm/filesystem/posix.cc:386:14: error: comparison of integer expressions of different signedness: ‘uint64_t’ {aka ‘long unsigned int’} and ‘long int’ [-Werror=sign-compare]
  386 |   if (offset > std::numeric_limits<off_t>::max()) {
      |       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[5]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/build.make:362: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/filesystem/posix.cc.o] Error 1
make[4]: *** [CMakeFiles/Makefile2:125: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
make[3]: *** [Makefile:141: all] Error 2
make[2]: *** [CMakeFiles/tiledb.dir/build.make:114: tiledb-prefix/src/tiledb-stamp/tiledb-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:308: CMakeFiles/tiledb.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

--- a/tiledb/sm/filesystem/posix.cc
+++ b/tiledb/sm/filesystem/posix.cc
@@ -383,7 +383,7 @@ Status Posix::read(
     return LOG_STATUS(Status::IOError(
         std::string("Cannot read from file; ") + strerror(errno)));
   }
-  if (offset > std::numeric_limits<off_t>::max()) {
+  if (offset > static_cast<uint64_t>(std::numeric_limits<off_t>::max())) {
     return LOG_STATUS(Status::IOError(
         std::string("Cannot read from file ' ") + path.c_str() +
         "'; offset > typemax(off_t)"));

